### ISSUE TITLE
`undefined` Value for Docs plugin `sidebarPath` Broken on Multi Version

### ISSUE THREAD

**Issue by nascode** [facebook/docusaurus#7968]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [ ] I have tried the `npm run clear` or `yarn clear` command.
- [ ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

According to https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-docs#sidebarPath if we set `undefined` as a value on `sidebarPath`, we can use fully autogenerated sidebar without using `sidebars.js` file.

On a fresh install, and set `sidebarPath` to `undefined` we can see that sidemenu autogenerated succesfully. However no sidemenu shown when enabling multiversioning.

### Reproducible demo

https://stackblitz.com/edit/github-bcjxsn

### Steps to reproduce

1. Create a new fresh instance of Docusaurus.
2. Open `docusaurus.config.js` set `sidebarPath` to undefined.
3. Delete `sidebars.js`. When you see the site, side menu will fully autogenerated.
4. Enable multiversioning, for example run `docusaurus docs:version 1.1.0` to add a version. 
5. Delete `versioned_sidebars` folder, sidebar will not be autogenerated.

### Expected behavior

Setting `undefined` as a value of `sidebarPath` Docs plugin config should be enough to show fully autogenerated side menu without `sidebars.js` even when we enable multiversioning.

### Actual behavior

When multiversioning enabled, no side menu is autogenerated. User should add `versioned_sidebars` manually even if we set `undefined` as a value of `sidebarPath`.

### Your environment

- Docusaurus version used: 2.0.1
- Environment name and version (e.g. Chrome 89, Node.js 16.4):
- Operating system and version: MacOS Monterey 12.5

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by Josh-Cena** [comment]
I never figured out how sidebar generation for versioned docs works. But last time I checked the logic indeed would fail in this case.

**Comment by slorber** [comment]
Yes indeed, on versioning even if `sidebarPath: undefined` is set, running the versioning cli would still generate a sidebar file in `versioned_sidebars`, and all "versioned versions" (not `./docs`) are expected to have a `sidebars.json` file.

Now, you'll notice that the emitted `sidebars.json` file is configured by default to provide this autogeneration:

```js
{
  "defaultSidebar": [
    {
      "type": "autogenerated",
      "dirName": "."
    }
  ]
}
```

What I understand here is that the bug is only because you deleted the `versioned_sidebars`: keep this folder and it will work fine.

Now I understand that if all your site and versions are using auto-generated sidebars, you may prefer to remove the useless `versioned_sidebars` folder to avoid polluting your site with useless files that all contain the same config?


---

What if someday you want to add an explicit sidebar for an older version? Would you remember that you have to manually create the `versioned_sidebars/version-1.1.0-sidebars.json` file to do so? Wouldn't it be more convenient if the file was already there?

For this reason, I think we should keep emitting the versioned sidebars files when using the versioning cli, even if `sidebarPath: undefined`.

Now, we should probably allow you to delete the folder if you think it's useless for your use-case to keep it.

What do you think?

**Comment by nascode** [comment]
> Now I understand that if all your site and versions are using auto-generated sidebars, you may prefer to remove the useless versioned_sidebars folder to avoid polluting your site with useless files that all contain the same config?

@Josh-Cena Yes, this is what I want to achieve. In my case, I am creating new version of docs manually (without running e.g. `npm run docusaurus docs:version 1.1.0`).

I am trying to avoid contributor add `versioned_sidebars` sidebar file to minimize confusion to those unfamiliar with Docusaurus internal.

> Now, we should probably allow you to delete the folder if you think it's useless for your use-case to keep it.
Yes, I think it would be better to make `undefined` `sidebarPath` behavior consistent on single and multi versioned docs. That is to use fully autogenerated sidebar by default.
